src/main/java/race/RacingCar.java

```diff
@@ -0,0 +1,52 @@
+package race;
+
+public class RacingCar {
+
+    private int position = 0;
+    private String name;
+    private static final int MOVE_CONDITION_NUMBER = 4;
+    private NumberPicker numberPicker;
+
+    public RacingCar(String name, NumberPicker numberPicker) {
+        validateNameLength(name);
+
+        this.name = name;
+        this.numberPicker = numberPicker;
+    }
+
+    public RacingCar(String name) {
+        this(name, new NumberPicker());
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public void moveForward() {
+        if (isMovable(numberPicker.generateRandomNumberUnderTen())) {
+            this.position++;
+        }
+    }
+
+    public int getPosition() {
+        return this.position;
+    }
+
+    public boolean checkSamePosition(int position) {
+        return this.position == position;
+    }
+
+    private boolean isMovable(int number) {
+        return number >= MOVE_CONDITION_NUMBER;
+    }
+
+    private void validateNameLength(String name) {
+        if (name.length() == 0) {
+            throw new IllegalArgumentException("자동차 이름은 빈 문자열일 수 없습니다.");
+        }
```

Comment:
name이 null 또는 공백문자일 수 있으니 더 체크할 수 있어야할 것 같아요

```suggestion
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("자동차 이름은 빈 문자열일 수 없습니다.");
        }
```

Reply:
Calculator에서 했던 걸 망각했네요.. 좀 더 고심해서 코드 짜도록 하겠습니다!